### PR TITLE
add algebra.js@0.2.6 w/ git auto-update

### PR DESCRIPTION
**Pull request by menuka94**
Pull request for issue: #9993

# Profile of the lib
 * Git repository (required): https://github.com/nicolewhite/algebra.js
 * NPM package url (optional): https://www.npmjs.com/package/algebra.js
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 1031
   - Count of watchers: 39
   - Count of forks: 44
 * NPM download stats (optional):
   - Downloads in the last day: 3
   - Downloads in the last week: 28
   - Downloads in the last month: 200

# Essential checklist
 * [ ] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [ ] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/algebra.js/package.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/nicolewhite/algebra.js.git"
+  },
+  "files": [
+    "algebra.js",
+    "src/"
+  ],
+  "autoupdate" : {
```
Comment (kennynaoh) [inline ajax/libs/algebra.js/package.json]: I think it's better to use npm auto-update because there are more complete releases on npm package.

Comment (kennynaoh) [inline ajax/libs/algebra.js/package.json]: Please take a look at this https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md

Comment (kennynaoh) [inline ajax/libs/algebra.js/package.json]: I think you can use two `npmFileMap` but different `basePath` to mention the file. One is `""` to grab the `algebra.js`, another is `"src"` to grab the files under `src/`, thanks!

File: ajax/libs/algebra.js/package.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "algebra.js",
+  "version": "0.2.6",
```
Comment (kennynaoh) [inline ajax/libs/algebra.js/package.json]: The latest version author published is `0.2.5`.

Comment (menuka94) [inline ajax/libs/algebra.js/package.json]: Could you please help me pull the commits with sparseCheckout? I tried but it doesn't seem to work.

**Comment by menuka94** [comment]
@x09326 I did not add those files. Those were from one of my previous contributions. I have no idea how those commits were merged into this PR.

**Comment by PeterDaveHello** [comment]
@menuka94 are you working locally now? We are happy to help solving this problem.

**Comment by menuka94** [comment]
@PeterDaveHello Yes. I usually work with Git command line. I tried using it for my last PR as well. But I couldn't clone the cdnjs repo. Then I came across the sparseCheckout method. However, I'm afraid I don't fully understand that method. Shouldn't I create a new branch when adding a new library? If so, how do I do the 
`$ echo '/ajax/libs/jquery/*' >> .git/info/sparse-checkout` step with it. I mean, should I first upload the files using GitHub GUI and then pull it into the local repo ?? I'd like to clarify these things. Your instructions are much appreciated. Thanks!

**Comment by PeterDaveHello** [comment]
@menuka94 happy to know that you are using the command line, hope we can learn something from this PR.

Once you are working with the command line, then the only two things on GitHub would be
  a. Open pull request
  b. Comment (discuss)

So actually, don't care about GitHub at this part anymore, you just need to update the sparseCheckout config like, `echo '/ajax/libs/libraryName/*' >> .git/info/sparse-checkout`, in this case: the string should be `/ajax/libs/algebra.js/*`, and then create a new branch for the library, you don't need to upload its files on GitHub, just commit them locally.

**Comment by PeterDaveHello** [comment]
@menuka94 let's squash these 9 commits into only one commit since most of them are fixes of the previous commits which is meaningless to the whole repository commit history, the command should be like this:
`git rebase -i HEAD~9`

it'll open a text editor for you, then just replace the `pick`s in the 2nd~9th lines, with `fixup` or `f`, then save and exit, git will help you squash them into only one commit.

@pvnr0082t please also take care of this PR!

**Comment by menuka94** [comment]
@PeterDaveHello Ok thanks. I'll do that.

**Comment by PeterDaveHello** [comment]
@menuka94 btw, please also rebase your branch with our latest master branch, your branch is a little bit old, thanks

**Comment by menuka94** [comment]
@PeterDaveHello I'll open another PR with the correct files. There were some issues when pulling the branch to my local repo.

**Comment by PeterDaveHello** [comment]
@menuka94 No problem, just don't forget to mention this old PR in the new one, thanks.

**Pull request closed without merging**